app/helpers/cases_helper.rb

```diff
@@ -263,6 +263,18 @@ def case_details_for_link_type(link_type)
     end
   end
 
+  def case_late_team_name(kase)
+    if kase.responded_late?
+      if kase.late_team.nil?
+        'Unspecified'
+      else
+        kase.late_team.name
+      end
+    else
+      'N/A'
+    end
+  end
+
```

Comment:
Agree with decision to move the `late_team_name` code - is there a decorator this function could belong to rather than a helper as that seems to be the way.

Reply:
This was in a decorator. It seems that the decorator 'pattern' breaks down when there are multiple (different) display formats for the same model.